pkg/metricscollector/opentelemetry.go

```diff
@@ -33,17 +33,18 @@ var (
 	otTriggerRegisteredTotalsCounter api.Int64UpDownCounter
 	otCrdRegisteredTotalsCounter     api.Int64UpDownCounter
 
-	otelScalerMetricVal                   OtelMetricFloat64Val
-	otelScalerMetricsLatencyVal           OtelMetricFloat64Val
-	otelScalerMetricsLatencyValDeprecated OtelMetricFloat64Val
-	otelInternalLoopLatencyVal            OtelMetricFloat64Val
-	otelInternalLoopLatencyValDeprecated  OtelMetricFloat64Val
+	otelScalerMetricVals                  []OtelMetricFloat64Val
+	otelScalerMetricsLatencyVals          []OtelMetricFloat64Val
+	otelScalerMetricsLatencyValDeprecated []OtelMetricFloat64Val
+	otelInternalLoopLatencyVals           []OtelMetricFloat64Val
+	otelInternalLoopLatencyValDeprecated  []OtelMetricFloat64Val
```

Comment:
In the previous implementation, these types of metrics would be called asynchronously in a callback by the OTLP library, which has its own internal. However, if two or more metrics were pushed in a short amount of time, the later one would replace the previous one, causing the previous metric to not be sent successfully. Therefore, I changed them to be stored in a slice to avoid this issue.

 I added a unit test to check this situation. In the previous implementation, this test will be failed and the first metric cannot be retrieved.
![image](https://github.com/kedacore/keda/assets/4278354/27426d5d-8262-41aa-89d5-5fc8a3970e4c)

Reply:
Interesting, good catch!